src/librustc_front/lowering.rs

```diff
@@ -1200,9 +1201,57 @@ pub fn lower_expr(lctx: &LoweringContext, e: &Expr) -> P<hir::Expr> {
             ExprIndex(ref el, ref er) => {
                 hir::ExprIndex(lower_expr(lctx, el), lower_expr(lctx, er))
             }
-            ExprRange(ref e1, ref e2) => {
-                hir::ExprRange(e1.as_ref().map(|x| lower_expr(lctx, x)),
-                               e2.as_ref().map(|x| lower_expr(lctx, x)))
+            ExprRange(ref e1, ref e2, lims) => {
+                fn make_struct(lctx: &LoweringContext,
+                               span: Span,
+                               path: Vec<&str>,
+                               fields: Vec<(&str, &P<Expr>)>) -> P<hir::Expr> {
```

Comment:
I don't think it matters here since I didn't construct a new `P`? It's just a matter of doing `&**e` within this utility function or on every call below.